code/modules/organs/subtypes/parasite.dm

```diff
@@ -0,0 +1,209 @@
+
+	parent_organ = "head"
+
+/obj/item/organ/parasite/blackkois/process()
+	..()
+
+
+	if(stage >= 3 && !(all_languages[LANGUAGE_VAURCA] in owner.languages))
+		set_light(-1.5, 6, "#FFFFFF")
+		owner.add_language(LANGUAGE_VAURCA)
+		owner << "<span class='notice'> Your mind expands, and your thoughts join the unity of the Hivenet.</span>"
+
+	if(stage >= 4)
+
+		var/obj/item/organ/brain/B = owner.internal_organs_by_name["brain"]
+
+		if(B && !B.lobotomized)
+			owner << "<span class='danger'>As the K'ois consumes your mind, you feel your past self, your memories, your very being slip away... only slavery to the swarm remains...</span>"
+			owner << "<b>You have been lobotomized by K'ois infection. All of your previous memories up until this point are gone, and all of your ambitions are nothing. You live for only one purpose; to serve the Lii'dra hive.</b>"
+
+			B.lobotomized = 1
+
+		for(var/datum/language/L in owner.languages)
+			owner.remove_language(L.name)
+		owner.add_language(LANGUAGE_VAURCA)
+
+
+	if(stage >= 1 && prob(10) && !(owner.species.flags & NO_PAIN))
+		if(stage < 3)
+			owner << "<span class='warning'>You feel a stinging pain in your abdomen!</span>"
+		else
+			owner << "<span class='warning'>You feel a stinging pain in your head!</span>"
+		owner.emote("me",1,"winces slightly.")
+		owner.adjustHalLoss(5)
+		return
+
+	if(stage >= 2)
+		if(stage < 3 && prob(10) && !(owner.species.flags & NO_BREATHE))
+			owner.emote("me", 1, "gasps for air!")
+			owner.losebreath += 5
+			return
+
+		else if(stage >= 3 && prob(10))
+			owner << "<span class='warning'>You feel disorientated!</span>"
+			switch(rand(1,3))
+				if(1)
+					owner.confused += 10
+					owner.apply_effect(10,EYE_BLUR)
+				if(2)
+					owner.slurring += 30
+				if(3)
+					owner.make_dizzy(10)
+			return
+
+	if(stage >= 3)
+		if(prob(5))
+			owner << "<span class='warning'>You feel something squirming inside of you!</span>"
+			owner.reagents.add_reagent("phoron", 4)
+			return
+
+	if(stage >= 4)
+
+		if(prob(10))
+			if(!(owner.species.flags & NO_PAIN))
+				owner << "<span class='warning'>You feel an unbearable pain in your mind!</span>"
+				owner.emote("scream")
+			owner.adjustBrainLoss(1)
+			return
+
+		if(prob(10))
+			owner << "<span class='danger'>You feel something alien coming up your throat!</span>"
+
+			var/turf/T = get_turf(owner)
+
+			var/datum/reagents/R = new/datum/reagents(100)
+			R.add_reagent("blackkois",10)
+			R.add_reagent("phoron",10)
+			var/datum/effect/effect/system/smoke_spread/chem/spores/S = new("blackkois")
+
+			S.attach(T)
+			S.set_up(R, 20, 0, T, 40)
+			S.start()
+
+			if(!(owner.species.flags & NO_PAIN))
+				owner.emote("scream")
+				owner.adjustHalLoss(15)
+				owner.drip(15)
+				owner.delayed_vomit()
+			return
+
+/obj/item/organ/parasite/blackkois/removed(var/mob/living/carbon/human/target)
+	if(all_languages[LANGUAGE_VAURCA] in target.languages && stage >= 3)
+		target.remove_language(LANGUAGE_VAURCA)
```

Comment:
I think that would leave them with no language at all, as all other languages are removed during stage 4.

Reply:
Correct. If the infection grows to stage 4 the only cure is cloning or cyborgification. By this point they've already been lobotomized by the K'ois.

Reply:
If that is intentional, then its fine.